crates/daphne/src/messages/mod.rs

```diff
@@ -240,28 +240,41 @@ impl ParameterizedDecode<DapVersion> for Extension {
 pub struct ReportMetadata {
     pub id: ReportId,
     pub time: Time,
+    pub public_extensions: Option<Vec<Extension>>,
 }
 
 impl ParameterizedEncode<DapVersion> for ReportMetadata {
     fn encode_with_param(
         &self,
-        _version: &DapVersion,
+        version: &DapVersion,
         bytes: &mut Vec<u8>,
     ) -> Result<(), CodecError> {
         self.id.encode(bytes)?;
         self.time.encode(bytes)?;
+        match (version, &self.public_extensions) {
+            (DapVersion::Draft09, None) => (),
+            (DapVersion::Latest, Some(extensions)) => {
+                encode_u16_items(bytes, version, extensions.as_slice())?;
+            }
+            _ => return Err(CodecError::UnexpectedValue),
```

Comment:
We usually use this for encoding. How about his:
```suggestion
            _ => return Err(CodecError::Other("encountered incorrectly set public extensions".into()))s
```